fvm/env.go

```diff
@@ -11,3 +26,120 @@ type Environment interface {
 	VM() *VirtualMachine
 	runtime.Interface
 }
+
+// Parts of the environment that are common to all transaction and script
+// executions.
+type commonEnv struct {
+	ctx           Context
+	sth           *state.StateHolder
+	vm            *VirtualMachine
+	programs      *handler.ProgramsHandler
+	accounts      state.Accounts
+	accountKeys   *handler.AccountKeyHandler
+	contracts     *handler.ContractHandler
+	uuidGenerator *state.UUIDGenerator
+	metrics       *handler.MetricsHandler
+	logs          []string
+	rng           *rand.Rand
+	traceSpan     opentracing.Span
+}
+
+func (env *commonEnv) Context() *Context {
+	return &env.ctx
+}
+
+func (env *commonEnv) VM() *VirtualMachine {
+	return env.vm
+}
+
+func (env *commonEnv) seedRNG(header *flow.Header) {
+	// Seed the random number generator with entropy created from the block
+	// header ID. The random number generator will be used by the UnsafeRandom
+	// function.
+	id := header.ID()
+	source := rand.NewSource(int64(binary.BigEndian.Uint64(id[:])))
```

Comment:
this should not work since 50% of time it will panic on overflowing the int64.

Reply:
does golang overflow / panic on this?

Reply:
Good pint! Go does not panic in runtime on overflow but instead fails to compile when it can detect the overflow:

this fails to compile:
```
_ = int64(math.MaxUint64)
```

while following works fine:
```
x := uint64(math.MaxUint64)
_ = int64(x)
```